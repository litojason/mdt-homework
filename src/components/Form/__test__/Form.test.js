import React from "react";
import renderer from "react-test-renderer";
// import { shallow, mount } from "enzyme";
// import "jest-enzyme";
import "jest-styled-components";

import {
  Form,
  InputContainer,
  Label,
  Input,
  Selection,
  TextArea,
} from "../index";

describe("Component: Form", () => {
  it("should render Form", () => {
    const component = <Form />;

    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render InputContainer", () => {
    const component = <InputContainer />;

    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render Label", () => {
    const component = <Label />;

    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render Input", () => {
    const component = <Input />;

    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render Selection", () => {
    const component = <Selection />;

    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render TextArea", () => {
    const component = <TextArea />;

    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
