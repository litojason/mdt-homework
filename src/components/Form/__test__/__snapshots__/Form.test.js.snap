// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Form should render 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<form
  className="c0"
/>
`;

exports[`Component: Form should render Form 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<form
  className="c0"
/>
`;

exports[`Component: Form should render Form 2`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<form
  className="c0"
/>
`;

exports[`Component: Form should render Form 3`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<form
  className="c0"
/>
`;

exports[`Component: Form should render Form 4`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<form
  className="c0"
/>
`;

exports[`Component: Form should render Input 1`] = `
.c0 {
  margin-bottom: 1rem;
  padding: 0.5rem 1rem;
  border-radius: 0.5rem;
  border-width: 1px;
}

<input
  className="c0"
/>
`;

exports[`Component: Form should render InputContainer 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
  className="c0"
/>
`;

exports[`Component: Form should render Label 1`] = `
.c0 {
  margin-bottom: 0.5rem;
}

<label
  className="c0"
/>
`;

exports[`Component: Form should render Selection 1`] = `
.c0 {
  margin-bottom: 1rem;
  padding: 0.5rem 1rem;
  border-radius: 0.5rem;
  border-width: 1px;
}

<select
  className="c0"
/>
`;

exports[`Component: Form should render TextArea 1`] = `
.c0 {
  resize: vertical;
  margin-bottom: 1rem;
  padding: 0.5rem 1rem;
  border-radius: 0.5rem;
  border-width: 1px;
}

<textarea
  className="c0"
/>
`;
